app/views/admin/badge_achievements/index.html.erb

```diff
@@ -0,0 +1,30 @@
+<div class="col">
+  <div class="row justify-content-end">
+    <%= link_to "Award Badge", admin_badge_achievements_award_badges_path, class: "btn btn-primary" %>
+  </div>
+</div>
+<table class="table">
+  <thead>
+    <tr>
+      <th scope="col">User</th>
+      <th scope="col">Badge</th>
+      <th scope="col">Badge Image</th>
+    </tr>
+  </thead>
+  <tbody>
+   <% @badge_achievements.includes([:user]).includes([:badge]).each do |badge_achievement| %>
+    <tr>
+      <td><%= badge_achievement.user.username %></td>
+      <td><%= badge_achievement.badge.title %></td>
+      <h5>
+        <td class="justify-content-center">
+          <img class="mx-auto mt-3" width="40" height="40" src="<%= badge_achievement.badge.badge_image %>" alt="badge image">
+        </td>
+        <td><%= link_to "Remove", url_for(action: :destroy, id: badge_achievement.id), method: :delete, data: { confirm: "Are you sure?" }, class: "btn btn-danger" %></td>
+      </h5>
+    </tr>
+    <% end %>
+  </tbody>
+</table>
+
+<%= paginate @badge_achievements %>
```

Comment:
I added pagination to this page because I am fearful that there will be a _ton_ of Badge Achievements on production, but I also fear that this isn't enough. Should I try to add a search bar of some sort in this view so that Admins can search via `username`?

Reply:
Hmm yeah I think a search function would be useful. I think probably `user_id` first is fine, since `username` might be a bit more complicated.

@jacobherrington had a good search implementation throughout `/admin` if you want to look at some examples. It's the [Ransack gem](https://github.com/activerecord-hackery/ransack).

Reply:
Okay, I'll take a look @Zhao-Andy! I actually halfway implemented search for this but ran into some issues and ended up switching gears since it wasn't part of the MVP. 😄